File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-0.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        # if kth largest then first element in a minHeap of size k
        minHeap = []
        for num in nums:
            heapq.heappush(minHeap,num)

            if len(minHeap) > k:
                heapq.heappop(minHeap)
        
        return minHeap[0]

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        /*
            1 2 3 2 2
            0 1 2 3 4
        */
        unordered_map<int,int>mp;
        for(auto i:nums){
            mp[i]++;
        }
        for(auto i:mp){
            if(i.second > 1){
                return i.first;
            }
        }
        return -1;
    }
};
